File: Data Structures & Algorithms/jump-game/submission-3.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int farthest = 0;
        for (int i = 0; i < n; i++) {
            if (i > farthest) return false;
            int leap = i + nums[i];
            farthest = Math.max(farthest, leap);
            if (farthest >= n - 1) return true;
        }
        return true;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        unordered_set<int> set;
        for (int num : nums) set.insert(num);

        int max = 0;
        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            if (!set.count(num - 1)) {
                int len = 0;
                while (set.count(num + len)) len++;
                max = std::max(max, len);
            }
        }
        return max;
    }
};
